### DISCUSSION TITLE
improve search effectiveness

### DISCUSSION THREAD

**Discussion by dsm1212** [TandoorRecipes/recipes#3088] [Ideas]
I frequently find myself searching and not finding what I want. I just don't have the keywords set. I've got a few suggested enhancements to improve searching in priority order:

1. automatically add every ingredient to search for every recipe (I think most of my search failures would be addressed with just this)
2. automatically add words from recipe name to search
3. search should AND the words I enter (or support AND/OR syntax as keywords)

**Comment by smilerz** [comment]
You can already modify search to do all of this.

**Comment by dsm1212** [comment]
Wow, I did not find this on the search settings until you responed. I thought that full text field was something else. I see there is a drop down there that includes ingredients and name.

I can't believe this has been there all along :-). Thanks!
